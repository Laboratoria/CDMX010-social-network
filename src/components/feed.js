export const feed = `
<div id="feedScreen" class="container2">
  <header class="menu">
      <img src="images/logo app2.png" class="logoMenu">
      <section class="iconHeader">
        <input type="image" id="alcaldiasIcon" class="alcaldiasIcon" src="images/alcaldias.png">
        <input type="image" id="menuIcon" class="menuIcon" src="images/menuIcon.png">
      </section>
  </header>
  <div id="btnPost">
    <input type="button" id="goPostScreen" class="orangeButton" value="CREAR NUEVO POST">
    </div>
  <div id="postContainer" class="postContainer"></div>
</div>
    `;

export const buildPost = async (onGetPost) => {
  const postContainer = document.querySelector('#postContainer');
  await onGetPost((postCollection) => {
    postContainer.innerHTML = '';
    // postCollection.then((doc) => {
    postCollection.forEach((element) => {
      const infoPost = element.data();
      infoPost.id = element.id;
      postContainer.innerHTML += `
      <div class="singlePost">
  <!--       <input type="image" id="profilePicture">
        <ul class="postData">
            <li>Nombre:</li>
            <li>Fecha:</li>
            <li>Ubicaci√≥n</li>
        </ul> -->
        <p>${infoPost.post}</p>
        <p>${infoPost.alcaldias}</p>
        <input type="image" id="likeIcon" class="likeIcon" src="images/likeIcon.png">
        <input type="image" id="editIcon" class="editIcon btnEdit" data-id="${infoPost.id}" src="images/editIcon.png">
        <input type="image" id="deleteIcon" class="deleteIcon btnDelete" data-id="${infoPost.id}" src="images/deleteIcon.png">
      </div>
        `;
    });
    // });
  });
};

export const removePost = /* async */ (deletePost, postId) => {
  // await deletePost(postId);
  deletePost(postId).then(() => {
    console.log(`post${postId}borrado con exito`);
  });
};
