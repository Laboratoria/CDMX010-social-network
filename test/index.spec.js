// importamos la funcion que vamos a testear
import { buildPost } from '../src/components/feed.js';

describe('build post', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="root"></div>';
  });
  test('debería renderizar', async () => {
    const target = document.getElementById('root');
    const postDB = [
      {
        alcaldias: 'obregon',
        id: 'G6cHwWgAOGnfidMokUeJ',
        like: '["susana23@gmail.com"]',
        post: 'hola',
      },
    ];
    const onGetPost = jest.fn().mockImplementation(() => Promise.resolve(postDB));
    //const firebase = { onGetPost };
    // console.log(onGetPost);
    await buildPost(onGetPost);
    expect(target.innerHTML).toMatchSnapshot();
  });
  // it('debería ser una función', () => {
  //   expect(typeof buildPost).toBe('function');
  // });
});
