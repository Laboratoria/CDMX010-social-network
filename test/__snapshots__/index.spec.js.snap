// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build post deber√≠a renderizar 1`] = `""`;
